packages/blockchain-wallet-v4-frontend/src/modals/Xlm/SunRiverWelcome/index.js

```diff
@@ -33,9 +33,28 @@ const Container = styled.div`
 `
 
 class SunRiverWelcomeContainer extends React.PureComponent {
+  // Only for users that are created but have not finished verification
+  continueVerification = () => {
+    this.props.modalActions.closeModal()
+    this.props.identityVerificationActions.verifyIdentity()
+  }
+
+  // Only for new users that have not started verification
+  goToIdentityVerification = () => {
+    this.props.modalActions.closeModal()
+    this.props.identityVerificationActions.createRegisterUserCampaign()
+  }
+
+  // Only for users who have completed/pending/under_review verification
+  viewStellarWallet = () => {
+    this.props.modalActions.closeModal()
+    this.props.routerActions.push('/xlm/transactions')
+  }
+
   determineKycState (userState, kycState) {
+    // always register the campaign as quickly as possible if user has been created
     if (userState !== USER_ACTIVATION_STATES.NONE) {
-      this.props.registerSunRiverUser()
+      this.props.identityVerificationActions.registerUserCampaign()
```

Comment:
ideally we should move this to `componentDidUpdate` and `componentDidMount` as we don't want to fire actions on each render, can b done in a future update